### ISSUE TITLE
DOM lifecycle events selection

### ISSUE THREAD

**Issue by janat08** [cyclejs/cyclejs#877]
Would be cool of if you could select lifecycle hooks for DOM driver, for if you're using jquery plugins. My solution to this was basically creating global events.

**Comment by jvanbruegge** [comment]
Have you looked into snabbom hooks? They allow you to do this

**Comment by janat08** [comment]
That's what I want to select.

**Comment by jvanbruegge** [comment]
Hooks just do something, there is nothing you could select, what exactly do you mean?

**Comment by janat08** [comment]
Create a bunch of streams for each node, and once a hook is run produce event for that stream? An update hook is arguably an event.
```
DOMSource.select('input').hooks('update' ...
```

**Comment by jvanbruegge** [comment]
yeah, an update is an event, but you already get updates with `DOMSource.select('input').element()`. I don't see the other hooks as events

**Comment by janat08** [comment]
The imperative dom libraries usually emit events with callbacks themselves. While its simply matters of hoisting to convert them into streams and use them in main, but it's uglier. Is it really difficult to plumb through the hooks to dom driver?

**Comment by janat08** [comment]
When I was referring to using global events above, I was referring to chaining onto jquery events rather than the hooks.

**Comment by jvanbruegge** [comment]
Honestly I still have problems understanding what you want to do in the end

**Comment by janat08** [comment]
You make an instance of widget for example, and that widget emits events for when it does something. What I'm complaining about is that your view function that will contain a bunch of event handlers, as opposed to them being in intents function.

**Comment by staltz** [comment]
@janat08 Could you try `DOMSource.select('input').element()` as jvanbruegge suggested?

**Comment by janat08** [comment]
If that updates a couple of times, it will instantiate several widgets.

**Comment by fuunnx** [comment]
Is https://github.com/cyclejs/cyclejs/issues/807 related to your question ?

**Comment by janat08** [comment]
No, I want the hooks in dom driver.
